### ISSUE TITLE
The default parameter of showCupertinoModalPopup() is inconsistent with the documentation

### ISSUE THREAD

**Issue by w568w** [flutter/flutter#96897]
In `packages/flutter/lib/src/cupertino/route.dart`, it declares the default value of parameter `useRootNavigator` as:
```dart
/// The `useRootNavigator` argument is used to determine whether to push the
/// popup to the [Navigator] furthest from or nearest to the given `context`. It
/// is `false` by default.
```

However, the signature of the method says:
```dart
Future<T?> showCupertinoModalPopup<T>({
  required BuildContext context,
  required WidgetBuilder builder,
  ImageFilter? filter,
  Color barrierColor = kCupertinoModalBarrierColor,
  bool barrierDismissible = true,
  bool useRootNavigator = true,
  bool? semanticsDismissible,
  RouteSettings? routeSettings,
})
```

Is it necessary to make `useRootNavigator` true by default? 

As a comparison, in the Material version of this method(`showModalBottomSheet`), the default value is also `false`. It makes an inconsistence and breaks some multi-platform invocations.

**Comment by danagbemava-nc** [comment]
The documentation for `showCupertinoDialog` also states: 

```dart
/// The `useRootNavigator` argument is used to determine whether to push the
/// dialog to the [Navigator] furthest from or nearest to the given `context`.
/// By default, `useRootNavigator` is `true` and the dialog route created by
/// this method is pushed to the root navigator.
```
and this is the method definition

```dart
Future<T?> showCupertinoDialog<T>({
  required BuildContext context,
  required WidgetBuilder builder,
  String? barrierLabel,
  bool useRootNavigator = true,
  bool barrierDismissible = false,
  RouteSettings? routeSettings,
}) {
  assert(builder != null);
  assert(useRootNavigator != null);

  return Navigator.of(context, rootNavigator: useRootNavigator).push<T>(CupertinoDialogRoute<T>(
    builder: builder,
    context: context,
    barrierDismissible: barrierDismissible,
    barrierLabel: barrierLabel,
    barrierColor: CupertinoDynamicColor.resolve(kCupertinoModalBarrierColor, context),
    settings: routeSettings,
  ));
}
```

It would be best if this was cleared up for users.

**Comment by piedcipher** [comment]
Currently the docs at `packages/flutter/lib/src/cupertino/route.dart` also uses `true`. Shall we close this issue?

**Comment by MitchellGoodwin** [comment]
> Currently the docs at `packages/flutter/lib/src/cupertino/route.dart` also uses `true`. Shall we close this issue?

You are correct. This was fixed in #106274. Closing.
